menuconfig.py

```diff
@@ -165,100 +165,266 @@
+
+        style = dict(_STYLES[sdef["style"]])
+    elif base:
+        style = dict(base)
+    else:
+        style = dict(_DEFAULT_STYLE)
 
-    # Initialize styles for different parts of the application. The arguments
-    # are ordered as follows:
-    #
-    #   1. Text color
-    #   2. Background color
-    #   3. Attributes
-    #   4. Extra attributes if colors aren't available. The colors will be
-    #      ignored in this case, and the attributes from (3.) and (4.) will be
-    #      ORed together.
+    # Merge the two dictionaries
+    for x in sdef:
+        # Skip the "style" special keyword
+        if x == "style": continue
+
+        if not x in style:
+            raise _StyleError("Unkown style keyword: {}".format(x))
 
-    # A_BOLD tends to produce faint and hard-to-read text on the Windows
-    # console, especially with the old color scheme, before the introduction of
-    # https://blogs.msdn.microsoft.com/commandline/2017/08/02/updating-the-windows-console-colors/
-    BOLD = curses.A_NORMAL if _IS_WINDOWS else curses.A_BOLD
+        style[x] = sdef[x]
 
-    # Default styling. Themes can override these settings below.
+    return style
 
-    # Top row in the main display, with the menu path
-    PATH_STYLE               = (curses.COLOR_BLACK, curses.COLOR_WHITE,  BOLD                              )
+def _style_add(name, sdef):
+    global _STYLES
+    style = _style_expand(sdef)
+    _STYLES[name] = style
 
-    # Separator lines between windows. Also used for the top line in the symbol
-    # information dialog.
-    SEPARATOR_STYLE          = (curses.COLOR_BLACK, curses.COLOR_YELLOW, BOLD,            curses.A_STANDOUT)
+def _style_from_string(sstr):
+    style = {}
 
-    # List of items, e.g. the main display
-    LIST_STYLE               = (curses.COLOR_BLACK, curses.COLOR_WHITE,  curses.A_NORMAL                   )
+    for sline in sstr.split(" "):
+        sline.strip(" ")
 
-    # Style for the selected item
-    LIST_SEL_STYLE           = (curses.COLOR_WHITE, curses.COLOR_BLUE,   BOLD,            curses.A_STANDOUT)
+        # Skip empty entries
+        if len(sline) == 0: continue
 
-    # Like _LIST_(SEL_)STYLE, for invisible items. Used in show-all mode.
-    LIST_INVISIBLE_STYLE     = (curses.COLOR_RED,   curses.COLOR_WHITE,  curses.A_NORMAL, BOLD             )
-    LIST_INVISIBLE_SEL_STYLE = (curses.COLOR_RED,   curses.COLOR_BLUE,   BOLD,            curses.A_STANDOUT)
+        skey = sline.split("=")
 
-    # Help text windows at the bottom of various fullscreen dialogs
-    HELP_STYLE               = PATH_STYLE
+        # Words without a "=" character imply a style selection
+        if len(skey) == 1:
+            # Implicity style selection
 
-    # Frame around dialog boxes
-    DIALOG_FRAME_STYLE       = SEPARATOR_STYLE
+            if not skey[0] in _STYLES:
+                raise _StyleError("Unknown built-in style: {}. Error near: {}".format(skey[0], sline))
```

Comment:
Could linebreak before the `.format()` to shorten the line a bit.

I try to stay within 80 columns.